### DISCUSSION TITLE
ImportError: cannot import name 'DAG' from 'airflow' (unknown location)

### DISCUSSION THREAD

**Discussion by eikesf** [apache/airflow#23598] [Q&A] [answered]
I've installed the airflow on docker and i'm trying to create my first DAG, but when i use the command `FROM airflow import DAG` and try to execute it gives an import error. The file name isn't set as airflow.py to avoid import problems. Also i can't import the `from airflow.operators.python_operator import PythonOperator` it says that the airflow.operators.python_operator could not be resolved.

Here's the code that i've used to create my first DAG:
```
import airflow
from airflow import DAG
from airflow.operators.python_operator import PythonOperator

default_args ={
    'owner': 'eike',
    'depends_on_past': False,
    'start-date': airflow.utils.dates.days_ago(2),
    'email': ['eike@gmail.com.br'],
    'email_on_failure': False,
    'email_on_retry': False,
    'retries': 2,
    'retry_delay': timedelta(minutes=3),
}

dag = DAG(
    'anonimização',
    default_args = default_args,
    description = 'Realização da anonimzação do banco de dados propesq',
    schedule_interval = timedelta(None),
    catchup = False,
)
```

After executing this code i get the issue:
```
Traceback (most recent call last):
  File "/home/eike-geosapiens/airflow/dags/anonimizar.py", line 3, in <module>
    from airflow import DAG
ImportError: cannot import name 'DAG' from 'airflow' (unknown location)
```
I'm currently using airflow 2.3.0 and pip 20.0.2 and python 3.8.10

**Comment by potiuk** [comment, marked as answer]
You must have installed airflow to a different virtualenv or something like that. This is more general python installation problem than airflow.  There are plenty things that you might have wrong - bad PYTHONPATH, differen user you use for running than for installation of airlfow are the first that come to mind - generally  - you need to debug your installation and runnning and you have to make sure you installed airflow in the same environment that you use for running it.

**Reply by eikesf** [reply]
Thank you, i'll try to solve it. That's my first time using airflow and i'm kinda lost. The best way is to create a virtualenv and install the python libraries and airflow inside of it ?

**Reply by potiuk** [reply]
Just follow the quick start docs https://airflow.apache.org/docs/apache-airflow/stable/start/index.html but if your job is to learn how to run and install python apps and need to learn it - to be perfectly honest this is not the pklace you shoudl ask for help. there are plenty of tutorials, courses to learn Python/virtualenv/installing apps and you should start from that. Airlfow is mostly a standard Python app but then it is rather complex to setup and manage. If you have no experience with setting up and managing python apps better use managed service like Astronomer, Composer MWAA. 

And read our docs first. You can see summary of all the ways you can install airflow here https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html together with expectations you should fulfill if you want to follow any of the routes. Please read that carefully to decide which road you want to follow (and make sure whoever ask you to do it also reads and understands it).
